src/topology/config/mod.rs

```diff
@@ -133,6 +135,7 @@ pub trait SinkConfig: core::fmt::Debug {
     fn build(
         &self,
         acker: crate::buffers::Acker,
+        dns: &DnsResolver,
```

Comment:
would it make more sense to pass a global options type of thing?

Reply:
Yes, I have gone from passing it like this, to passing `GlobalOptions`, then just to return to passing it directly. 

`GlobalOptions` was a good way to pass it, but with recent changes that parse `GlobalOptions` from `toml`, instead of constructing it as it was before, it doesn't make sense to have non data struct in a struct primarily meant for data.

And when it comes to creating some new global options that could be passed, I think it's outside of scope. It should be done on it's own with more thought put into it.